openverse_catalog/dags/maintenance/add_license_url.py

```diff
@@ -76,60 +78,85 @@ def update_license_url(postgres_conn_id: str, task: AbstractOperator) -> dict[st
-            WHERE (
-            meta_data is NULL AND license = '{license_name}'
-            AND license_version = '{license_version}')
-            """
-        )
-        result = postgres.get_records(select_query)
-
-        if not result:
-            logger.info(f"No records to update with {license_url}.")
+    select_query = dedent(
+        """
+    SELECT identifier, license, license_version
+    FROM image WHERE meta_data IS NULL;"""
+    )
+    records_with_null_in_metadata = postgres.get_records(select_query)
+
+    # Dictionary with license pair as key and list of identifiers as value
+    records_to_update = {}
+
+    for result in records_with_null_in_metadata:
+        identifier, license_, version = result
+        license_pair = f"{license_},{version}"
+        if license_pair not in records_to_update:
+            records_to_update[license_pair] = [identifier]
+        else:
+            records_to_update[license_pair].append(identifier)
+
+    total_updated = 0
+    updated_by_license = {}
+
+    for license_pair, identifiers in records_to_update.items():
+        license_, license_version = license_pair.split(",")
+        license_url = get_license_info_from_license_pair(license_, license_version)[-1]
+        if license_url is None:
+            logger.info(
+                f"No license pair {license_pair} in the license reverse path map."
+            )
             continue
-        logger.info(f"{len(result)} records to update with {license_url}.")
-        license_url_col = {"license_url": license_url}
-        update_license_url_query = dedent(
+        logger.info(f"{len(identifiers):4} items will be updated with {license_url}")
+        license_url_dict = {"license_url": license_url}
+        update_query = dedent(
             f"""
             UPDATE image
-            SET meta_data = {Json(license_url_col)}
-            WHERE identifier IN ({','.join([f"'{r[0]}'" for r in result])});
+            SET meta_data = {Json(license_url_dict)}
+            WHERE identifier IN ({','.join([f"'{r}'" for r in identifiers])});
             """
         )
-
-        updated_count = postgres.run(
-            update_license_url_query, autocommit=True, handler=RETURN_ROW_COUNT
+        updated_count: int = postgres.run(
+            update_query, autocommit=True, handler=RETURN_ROW_COUNT
         )
         logger.info(f"{updated_count} records updated with {license_url}.")
-        total_counts[license_url] = updated_count
-        total_count += updated_count
-
-    logger.info(f"{total_count} image records with missing license_url updated.")
-    for license_url, count in total_counts.items():
-        logger.info(f"{count} records with {license_url}.")
-    return total_counts
+        if updated_count:
+            updated_by_license[license_url] = updated_count
+        total_updated += updated_count
+    logger.info(f"Updated {total_updated} rows")
+    return json.dumps(updated_by_license)
```

Comment:
You should be able to return this directly without dumping it as a JSON string before hand! You'll just need to set `render_template_as_native_obj=True,` in the DAG settings ([here's an example](https://github.com/WordPress/openverse-catalog/blob/af205795b412da9f3d0d7f6d7e066b7d039b1763/openverse_catalog/dags/maintenance/pr_review_reminders/pr_review_reminders_dag.py#L52)). Then you can do a standard `None` check/use the dictionary directly in the `final_report` task below 🙂

Reply:
Thank you for the suggestion, that is so much better :)